http_requests.py

```diff
@@ -221,3 +226,66 @@ def delete(self, url: str, return_type=None, **kwargs):
                 return_type=None,
                 **kwargs
         )
+
+
+class EncodingMethod(enum.StrEnum):
+    GZIP = 'gzip'
+
+
+def encode_request(
+    data: str | bytes | dict | typing.IO=None,
+    json: dict=None,
+    headers: dict[str, str]=None,
+    encoding_method: EncodingMethod=EncodingMethod.GZIP,
+) -> (
+    tuple[collections.abc.Generator[bytes, None, None], dict[str, str]]
+    | collections.abc.Generator[bytes, None, None]
+):
+    '''
+    Encodes the given `data` or `json` property based on the selected `encoding_method`. If both
+    `data` and `json` are provided, the `data` property is used (same behaviour as implemented by
+    the `requests` library).
+
+    The corresponding `Content-Encoding` header is patched-in to the provided headers dictionary.
+
+    If `headers` is provided and not `None`, the response is a tuple of the compression result and
+    the patched headers, otherwise only the compression result is returned.
+    '''
+    if not data and json is not None:
+        data = js.dumps(json)
+
+    def _encode(obj, encoding: str='utf-8') -> bytes:
+        if isinstance(obj, bytes):
+            return obj
+        elif isinstance(obj, str):
+            return obj.encode(encoding=encoding)
+        elif isinstance(obj, dict):
+            return js.dumps(obj).encode(encoding=encoding)
+        else:
+            raise NotImplementedError(f'Encoding of type {type(obj)} is not (yet) supported')
+
+    def _compress(data, encoding_method) -> collections.abc.Generator[bytes, None, None]:
+        if isinstance(data, io.BufferedIOBase):
+            if encoding_method == EncodingMethod.GZIP:
```

Comment:
nit: enum-values ought to be compared using identity (`is` instead of `==`)

Reply:
You're right, you can, however, compare string enums directly to strings using `==`. If you're using `is` for comparison, you have to compare against `enum.value`.

Reply:
Of course I can change it, if you'd prefer the latter?